File: 2600-k-items-with-the-maximum-sum/2600-k-items-with-the-maximum-sum.ts
function kItemsWithMaximumSum(numOnes: number, numZeros: number, numNegOnes: number, k: number): number {
    if(numOnes >= k)    return k;
    if(numOnes + numZeros >= k) return numOnes;
    return numOnes - (k-numOnes-numZeros); 
};

File: 2600-k-items-with-the-maximum-sum/2600-k-items-with-the-maximum-sum.cpp
class Solution {
public:
    int kItemsWithMaximumSum(int numOnes, int numZeros, int numNegOnes, int k) {
        
        if(numOnes > k){
            return k;
        }else if(numOnes + numZeros >= k){
            return numOnes;
        }else if(numOnes + numZeros + numNegOnes >= k){
            int num = k - (numOnes + numZeros);
            return numOnes - num;
        }
        
        return 0;
        
    }
};